plugins/providers/virtualbox/action/network.rb

```diff
@@ -258,30 +258,44 @@ def hostonly_config(options)
           # Default IP is in the 20-bit private network block for DHCP based networks
           options[:ip] = "172.28.128.1" if options[:type] == :dhcp && !options[:ip]
 
-          # Calculate our network address for the given IP/netmask
-          netaddr  = network_address(options[:ip], options[:netmask])
-
-          # Verify that a host-only network subnet would not collide
-          # with a bridged networking interface.
-          #
-          # If the subnets overlap in any way then the host only network
-          # will not work because the routing tables will force the
-          # traffic onto the real interface rather than the VirtualBox
-          # interface.
-          @env[:machine].provider.driver.read_bridged_interfaces.each do |interface|
-            that_netaddr = network_address(interface[:ip], interface[:netmask])
-            raise Vagrant::Errors::NetworkCollision if \
-              netaddr == that_netaddr && interface[:status] != "Down"
-          end
+          ip = IPAddr.new(options[:ip])
+          if !ip.ipv6?
```

Comment:
Should probably do the `v4/v6` switch consistently in the places we do it.

I think the one below is a bit clearer:

```ruby
if ip.ipv4?
  # ...
elsif ip.ipv6?
  # ...
else
  raise "Malformed IPAddr: #{ip.inspect}"
end
```